### ISSUE TITLE
Lack of validation on eks.NodeGroup's ignore_changes string values

### ISSUE THREAD

**Issue by antifuchs** [pulumi/pulumi#7081]
We just ran into a very tooth-grinding problem: we'd set `ignore_changes=["scaling_config.desired_size"]` on an [eks.NodeGroup](https://www.pulumi.com/docs/reference/pkg/aws/eks/nodegroup/). The problem is that this string is not valid - it should be `["scalingConfig.desiredSize"]`, but pulumi didn't reject the value and so it sat, for some weeks.

When we updated an unrelated node group setting yesterday, pulumi changed the desired_size setting back to what was configured. It was a bit of an unpleasant surprise to find 23 nodes suddenly getting terminated & spun up again.

I have two suggestions that would help avoid this problem:
* Validate the string values on `ignore_changes`. Presumably there is a finite amount of them, so pulumi should ensure we pass the right values before throwing out invalid strings.
* Throw away `ignore_changes` entirely and let us specify values that should only ever be initialized and never re--set with a type.

Either of these would increase our confidence in our pulumi-defined infra a lot!

**Comment by lukehoban** [comment]
> Validate the string values on ignore_changes. Presumably there is a finite amount of them, so pulumi should ensure we pass the right values before throwing out invalid strings.

Yes - this is a very good idea.  We could at least warn if the ignoreChanges string doesn't match any property on the resource.

> Throw away ignore_changes entirely and let us specify values that should only ever be initialized and never re--set with a type.

What you describe is effectively what `ignoreChanges` does already - do you have more details on what you were hoping Pulumi might do here?

**Comment by benesch** [comment]
> > Throw away ignore_changes entirely and let us specify values that should only ever be initialized and never re--set with a type.
> 
> What you describe is effectively what `ignoreChanges` does already - do you have more details on what you were hoping Pulumi might do here?

I believe the problem arises when using `ignoreChanges` to ignore a subfield. Specifically, an EKS node group that says to ignore the desired size

```
aws.eks.NodeGroup(
    f"ng",
    cluster_name="cluster",
    scaling_config=aws.eks.NodeGroupScalingConfigArgs(
        desired_size=1,
        max_size=20,
        min_size=1,
    ),
    opts=ResourceOptions(ignore_changes=["scalingConfig.desiredSize"]),
)
```

will correctly ignore the cluster-autoscaler's _changes_ to `scalingConfig.desiredSize`, if you don't touch `min_size` or `max_size`. But if you update either `min_size` or `max_size`, Pulumi will update the entire scaling config, resetting `desired_size` to 1.

**Comment by lukehoban** [comment]
> But if you update either min_size or max_size, Pulumi will update the entire scaling config, resetting desired_size to 1.

This doesn't sound expected, and I can't repro this myself for an "update".  If the resource is being replaced though (a creation and a deletion), then the creation of the new resource will indeed use the default values provided in the resource definition, as there is no prior value to apply.  It's possible we could support a model where replacements also use previous values for ignored changes - but where the previous value is taken from the previous resource.  Do you know if that's the specific case you are looking for here?

Deploying this:

```ts
const bucket = new aws.s3.Bucket("my-bucket", {
    website: {
        indexDocument: "index.html",
        errorDocument: "err.html",
    }
});
```

Then updating it to this:

```ts
const bucket = new aws.s3.Bucket("my-bucket", {
    website: {
        indexDocument: "index.html",
        errorDocument: "err.html",
    }
}, { ignoreChanges: ["website.errorDocument"] });
```

Then finally to this:

```ts
const bucket = new aws.s3.Bucket("my-bucket", {
    website: {
        indexDocument: "index2.html",
        errorDocument: "err2.html",
    }
}, { ignoreChanges: ["website.errorDocument"] });
```

Results in:

```json
...
                    "website": {
                        "errorDocument": "err.html",
                        "indexDocument": "index2.html",
                        "redirectAllRequestsTo": "",
                        "routingRules": ""
                    },
...
```
